src/core/Akka.TestKit/TestKitBase_Receive.cs

```diff
@@ -383,20 +395,34 @@ public ValueTask<(bool success, MessageEnvelope envelope)> TryPeekOneAsync(TimeS
             if (maxDuration.IsZero())
             {
                 ConditionalLog(shouldLog, "Trying to peek message from TestActor queue. Will not wait.");
-                var didPeek = _testState.Queue.TryPeek(out var item);
+                var didPeek = _testState.Queue.Reader.TryPeek(out var item);
                 peek = (didPeek, item);
             }
             else if (maxDuration.IsPositiveFinite())
             {
                 ConditionalLog(shouldLog, "Trying to peek message from TestActor queue within {0}", maxDuration);
-                peek = await _testState.Queue.TryPeekAsync((int)maxDuration.TotalMilliseconds, cancellationToken)
-                    ;
+                var delayTask = Task.Delay(maxDuration, cancellationToken);
+                var readTask = _testState.Queue.Reader.WaitToReadAsync(cancellationToken).AsTask();
```

Comment:
Have to do this instead of playing around with `CancellationToken`s expiring, because otherwise those work via throwing exceptions. This allows us to gracefully determine if we've hit the next point because stuff was ready or because time expired.